kinova_driver/include/kinova_driver/kinova_comm.h

```diff
@@ -0,0 +1,198 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL CLEARPATH ROBOTICS, INC. BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Please send comments, questions, or patches to skynet@clearpathrobotics.com
+ *
+ */
+
+#ifndef KINOVA_DRIVER_KINOVA_COMM_H
+#define KINOVA_DRIVER_KINOVA_COMM_H
+
+#include <boost/thread/recursive_mutex.hpp>
+
+#include <kinova/KinovaTypes.h>
+
+#include <kinova_driver/kinova_ros_types.h>
+#include "kinova_driver/kinova_api.h"
+#include <std_srvs/Trigger.h>
+#include "ovis_msgs/OvisJointVelocity.h"
+#include "ovis_msgs/OvisJointAngle.h"
+
+namespace kinova
+{
+class KinovaComm
+{
+public:
+  KinovaComm(const ros::NodeHandle& node_handle, boost::recursive_mutex& api_mutex, const bool is_movement_on_start,
+             const std::string& kinova_robotType);
+  ~KinovaComm();
+
+  // %Tag(general functions)
+
+  void stopAPI();
+  void startAPI();
+  bool isStopped();
+
+  int robotType() const;
+  void getQuickStatus(QuickStatus& quick_status);
+  void getConfig(ClientConfigurations& config);
+  void setConfig(const ClientConfigurations& config);
+  void printConfig(const ClientConfigurations& config);
+  void getControlType(int& controlType);  // joint or cartesian control
+  void getGeneralInformations(GeneralInformations& general_info);
+  void getSensorsInfo(SensorsInfo& sensor_Info);   // voltage, current, temperatue and accelerometers
+  void getForcesInfo(ForcesInfo& force_Info);      // joint torque and end-effector wrench
+  void getGripperStatus(Gripper& gripper_status);  // most complete information of each fingers, including model,
+                                                   // motion, force, limits, etc.
+  void homeArm(void);
+  bool isHomed(void);
+  void initFingers(void);
+  void setEndEffectorOffset(unsigned int status, float x, float y, float z);
+  void getEndEffectorOffset(unsigned int& status, float& x, float& y, float& z);
+
+  // %EndTag(general functions)
+
+  // %Tag(Angular Control)%
+
+  void setAngularControl();
+  void getAngularCommand(AngularPosition& angular_command);
+  void getJointAngles(KinovaAngles& angles);
+  void setJointAngles(const KinovaAngles& angles, double speedJoint123 = 20, double speedJoint4567 = 20,
+                      int timeout = 0, bool push = true);
```

Comment:
These are a lot of magic numbers / variables for the default values of this function.

Reply:
Won't change this since it's code from Kinova, and they're default values.